extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -552,8 +552,12 @@ export class ProjectsController {
 				model.database = profile.databaseName;
 			}
 			else {
-				const connectionId = (await this.apiWrapper.openConnectionDialog()).connectionId;
+				const connection = await this.apiWrapper.openConnectionDialog();
+				if (!connection) {
```

Comment:
just to confrrm - can this happen due to some error? or only when cancelled

Reply:
I think only when cancelled. If connection was successful it returns the connection otherwise returns undefined. If there was any error during connection I think the connection dialog will handle it